CMakeLists.txt

```diff
@@ -14,19 +14,15 @@ include_directories(${PYTHON_INCLUDE_DIR})
 
 #############################################################################################
 # set the default install prefix
-# option(INSTALL_USER "Install into python user site directory (like python setup.py install --user)" OFF)
-# option(INSTALL_SITE "Install into python system site directory (like python setup.py install)" OFF)
+SET(CMAKE_INSTALL_PREFIX ${PYTHON_SYSTEM_SITE} CACHE PATH "Python site installation directory" FORCE)
+message(STATUS "Setting installation site dir: ${CMAKE_INSTALL_PREFIX}")
 
-# IF(INSTALL_USER)
-#     SET(CMAKE_INSTALL_PREFIX ${PYTHON_USER_SITE} CACHE PATH "Python site installation directory" FORCE)
-#     message(STATUS "Setting installation site dir: ${CMAKE_INSTALL_PREFIX}")
-# ELSEIF(INSTALL_SITE)
-if (INSTALL_SITE)
-    SET(CMAKE_INSTALL_PREFIX ${PYTHON_SYSTEM_SITE} CACHE PATH "Python site installation directory" FORCE)
-    message(STATUS "Setting installation site dir: ${CMAKE_INSTALL_PREFIX}")
-ENDIF()
+if (WIN32)
+add_compile_definitions(_CRT_SECURE_NO_WARNINGS)
```

Comment:
Windows suggests using: `strncpy_s` and `_sopen_s`. I don't see how `_sopen_s` is any better than `_open`.

Clang-tidy suggests using: `memcpy_s`, `memset_s`, and `strncpy_s`. I don't see how passing the size argument twice to a `_s` variant increases security. Most of the places where gsd uses these methods are immediately preceded with a memory allocation or resizes. Those few places where it is not, I am using `strncpy` to prevent copying past the length of the fixed destination buffer. Additionally, the `_s` variants are in Annex K, an *optional* extension to the C11 standard. For portability, we would need to keep the existing non `_s` variant code path and use it when Annex K is not available.